ext/js/language/translator.js

```diff
@@ -342,6 +346,20 @@ export class Translator {
         return deinflections;
     }
 
+    /**
+     * @param {string} searchResolution
+     * @param {number} i
+     * @param {string} source
+     * @returns {number}
+     */
+    _getNextSubstringLength(searchResolution, i, source) {
+        if (searchResolution === 'word') {
+            return source.search(new RegExp('[^\\p{Letter}]([\\p{Letter}\\p{Number}]*)$', 'u'));
```

Comment:
Why did you opt for a Regex constructor as opposed to a Regex literal?

Reply:
No particular reason, they should be interchangeable

Reply:
According to the [MDN Web Docs RegEx guide](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_expressions):
> Regular expression literals provide compilation of the regular expression when the script is loaded. If the regular expression remains constant, using this can improve performance.
> Using the constructor function provides runtime compilation of the regular expression. Use the constructor function when you know the regular expression pattern will be changing, or you don't know the pattern and are getting it from another source, such as user input.
In this case, I think replacing `new RegExp('[^\\p{Letter}]([\\p{Letter}\\p{Number}]*)$', 'u')` with `\[^\p{Letter}]([\p{Letter}\p{Number}]*)$\u` is advisable.